src/librustc_errors/json.rs

```diff
@@ -417,8 +418,14 @@ impl DiagnosticCode {
                 DiagnosticId::Error(s) => s,
                 DiagnosticId::Lint(s) => s,
             };
-            let explanation =
-                je.registry.as_ref().and_then(|registry| registry.find_description(&s));
+            let reg_result =
+                je.registry.as_ref().and_then(|registry| Some(registry.find_description(&s)));
```

Comment:
I think this code'll be easier to read if we have an `if let Some(registry) = je.registry { ... }` and avoid the combinators on Option here.